internal/services/v1/relationships.go

```diff
@@ -148,12 +149,47 @@ func (ps *permissionServer) ReadRelationships(req *v1.ReadRelationshipsRequest,
 		DispatchCount: 1,
 	})
 
+	limit := 0
+	var startCursor options.Cursor
+
+	rrRequestHash, err := computeReadRelationshipsRequestHash(req)
+	if err != nil {
+		return shared.RewriteError(ctx, err)
+	}
+
+	if req.OptionalCursor != nil {
+		decodedCursor, err := cursor.DecodeToDispatchCursor(req.OptionalCursor, rrRequestHash)
+		if err != nil {
+			return shared.RewriteError(ctx, err)
+		}
+
+		if len(decodedCursor.Sections) != 1 {
+			return shared.RewriteError(ctx, NewInvalidCursorErr("did not find expected resume relationship"))
+		}
```

Comment:
is this so that you cannot use a LR cursor in this API? Shouldn't this be centralized into a validation method?

Reply:
The hash check above makes sure you don't send a cursor from LR, but I wanted to be extra careful to not cause a bounds check error, so I added this